src/deviceinfo.cpp

```diff
@@ -355,4 +358,47 @@ QString DeviceInfo::wlanMacAddress()
     return d->wlanMacAddress();
 }
 
+static QString normalizeUid(const QString &uid)
+{
+    // Normalize by stripping colons, dashes and making it lowercase
+    return uid.trimmed().replace(":", "").replace("-", "").toLower();
+}
+
+QString DeviceInfo::deviceUid()
+{
+    Q_D(DeviceInfo);
+    if (!d->m_synchronousInit) {
+        // would need to ensure we don't return anything until sure the imeis are fetched etc
+        // let's just start with simple and require the synchronous mode, which should be
+        // sufficient for now
+        qWarning() << "DeviceInfo::deviceUid only available on synchronous instances";
```

Comment:
Just thinking out loud: would it be better to return obviously invalid value here? Versus afaict one of the possible values below that can change over time (initially file based fallback, later imei/mac when async init finishes).

Reply:
Oh yes. Was even supposed to make it so, but apparently missed it.